src/main/java/cli/scrape.java

```diff
@@ -105,61 +160,83 @@ public void run() {
+    @CommandLine.Option(names = "--pretty", defaultValue = "false" ) private String pretty;
 
     public void run() {
       long start = System.nanoTime();
-      String school = this.school;
-      String subject = this.subject;
-      Term term = Term.fromId(Integer.parseInt(this.term));
+      Term term;
+      if (this.term == null && this.semester == null && this.year == null) {
+        throw new IllegalArgumentException(
+            "Must provide at least one. Either --term OR --semester AND --year");
+      } else if (this.term == null) {
+        if(this.semester == null || this.year == null) {
+          throw new IllegalArgumentException(
+                  "Must provide both --semester AND --year"
+          );
+        }
+        term = new Term(Semester.fromCode(this.semester), year);
+      } else {
+        term = Term.fromId(this.term);
+      }
       if (school == null) {
         if (subject != null) {
           throw new IllegalArgumentException(
               "--subject doesn't make sense if school is null");
         }
-        try {
-          UtilsKt.writeToFileOrStdout(
-              JsonMapper.toJson(Scrape_catalogKt.scrapeFromCatalog(
-                                    term, SubjectCode.allSubjects(), batchSize),
-                                Boolean.parseBoolean(pretty)),
-              outputFile);
-        } catch (IOException e) {
-          e.printStackTrace();
-        }
+        Scrape_catalogKt
+            .scrapeFromCatalog(term, SubjectCode.allSubjects(), batchSize)
+            .iterator()
+            .forEachRemaining(value -> {
+              try {
+                System.out.println(
+                    JsonMapper.toJson(value, Boolean.parseBoolean(pretty)));
+              } catch (JsonProcessingException e) {
+                logger.warn(e.getMessage());
+              }
+            });
       } else if (subject == null) {
-        try {
-          UtilsKt.writeToFileOrStdout(
-              JsonMapper.toJson(Scrape_catalogKt.scrapeAllFromCatalog(
-                                    term, school, batchSize),
-                                Boolean.parseBoolean(pretty)),
-              outputFile);
-        } catch (IOException e) {
-          logger.warn(e.getMessage());
-        }
+        Scrape_catalogKt.scrapeAllFromCatalog(term, school, batchSize)
+            .iterator()
+            .forEachRemaining(value -> {
+              try {
+                System.out.println(
+                    JsonMapper.toJson(value, Boolean.parseBoolean(pretty)));
+              } catch (JsonProcessingException e) {
+                e.printStackTrace();
+              }
+            });
       } else {
         if (batchSize != null) {
           throw new IllegalArgumentException(
               "Batch size doesn't make sense when only doing on query");
         }
         try {
-          UtilsKt.writeToFileOrStdout(
+          System.out.println(
```

Comment:
Could you change this back to `writeToFileOrStdout` ? We want to be able to output to file if necessary

Reply:
This will give null pointers exception because of the way it's done in Kotlin. I'll write one in Java

Reply:
Nvm, I figure out why. output file should be the first param.